open-craft/xblock-group-project-v2
# Mitigate XSS attacks by using jquery's HTML encoding on output

JavaScript within certain XBlocks was outputting raw HTML from jqHXR requests. These could possibly be controlled by end users, resulting in XSS if such a user was malicious.
This PR changes the jquery `html()` calls to `text()` which uses jquery's built in HTML encoding to prevent this kind of attack.

**JIRA tickets**: Implements OC-3222

**Testing instructions**:

1. Use group project to display a message containing HTML such as `<blink>TEST</BLINK>`. This can be accomplished by opening the console, and running `GroupProjectCommon.Messages.show_message("<b>message</b><script>alert('test);</script>", "title")`
2. Check that this HTML output is properly encoded, and not rendered in the client's browser.

**Author notes and concerns**:

1. It may be advisable to also filter input for HTML characters, but should not be necessary in this case.

**Reviewers**
